src/components/DialogStates.vue

```diff
@@ -0,0 +1,138 @@
+<script lang="ts">
+import { defineComponent, ref, computed, Ref } from 'vue';
+
+// Component used to render dialog window which you can fill with custom content
+// Allows to switch between states
+// Available states:
+// - default
+// - form
+// Available slots:
+// - title
+// - content
+// Exposed v-slot props
+// - state
+// - setState
+// - reset
+// The content and image are rendered side by side in a scrollable dialog window
+export default defineComponent({
+  name: 'DialogStates',
+  props: {
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  emits: ['update:modelValue', 'change'],
+  setup(props, { emit }) {
+    const STATES = ['default', 'form'];
+
+    const activeState: Ref<'default' | 'form'> = ref('default');
+
+    const dialogOpen = computed({
+      get: () => props.modelValue,
+      set: (value) => {
```

Comment:
Missing func input value param type and return value type set: (value: boolean): void => {

Reply:
Added.